### PR TITLE
loop.getaddrinfo accepts service names like "http" for port.

### PR DESCRIPTION
**Pull request by ajdavis**
loop.getaddrinfo accepts service names like "http" for port.

### CODE REVIEW

**Comment by ajdavis** [comment]
Fixes #353 .

File: asyncio/base_events.py [hunk 1]
```diff
@@ -104,8 +104,14 @@ def _ipaddr_info(host, port, family, type, proto):
...
         port = 0
-    elif isinstance(port, (bytes, str)):
+    elif isinstance(port, bytes):
         port = int(port)
+    elif isinstance(port, str):
+        try:
+            port = int(port)
+        except ValueError:
+            # Might be a service name like "http".
+            port = socket.getservbyname(port)
```
Comment (1st1) [inline asyncio/base_events.py]: This works only when `port` is an `str`, right?  How does the `port=b'http'` test pass?

Comment (ajdavis) [inline asyncio/base_events.py]: Good catch! Fixed.

Comment (1st1) [inline asyncio/base_events.py]: Ah, I see. But why don't we just convert bytes to str?

`socket.getaddrinfo('127.0.0.1', b'http')` works so should our code.

File: asyncio/base_events.py [hunk 2]
```diff
@@ -104,8 +104,18 @@ def _ipaddr_info(host, port, family, type, proto):
 
     if port in {None, '', b''}:
```
Comment (1st1) [inline asyncio/base_events.py]: While you're here, could you please fix a `BytesWarning`:

Please replace `if port in {None, '', b''}:` with `if port is None:`; 
add `if port == '': port = 0 else:` to the `elif isinstance(port, str)`, 
do the same for bytes.

Comment (ajdavis) [inline asyncio/base_events.py]: Done.

**Comment by 1st1** [comment]
Merged by hand in 2959a97823cdf2905e343efc7f3ec8dc013970f7

Thanks a lot, Jesse.

**Comment by 1st1** [comment]
See also https://hg.python.org/cpython/rev/3ec208c01418

**Pull request closed without merging**